### ISSUE TITLE
Cursor is stuck in bottom of client screen. Issue with high resolution displays?

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#94]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/94
> 
> Issue created on: 2018-07-12 by @mcx808
> Issue last updated on: 2021-11-04

### Operating Systems ###

Server: Windows 10 x64 1709

Client: macOS 10.13.5

### Barrier Version ###

2.1.0 

### Steps to reproduce bug ###

I have a Surface book (3000x2000 scaled at 200%) with a 3840x2160 monitor above scaled at 150%.
My MacbookPro (1280x800) is to the right. Whenever I move the mouse to the mac, the cursor jumps straight to the bottom right screen and stays there. I can right click but not move it. I can use the mac trackpad to move the mouse but not the Windows mouse.

I disabled my laptop screen so I am only using the external display. Same issue.

I moved the orientation of the mbp to the right. Same issue.

The log shows the following warning when I move to the client screen: WARNING: cursor may not be visible

### Other info ###

* When did the problem start to occur? Every time the above steps are followed
* Is there a way to work around it? No
* Does this bug prevent you from using Barrier entirely? Yes

Logs are attached
[clienttlog.txt](https://github.com/debauchee/barrier/files/2188116/clienttlog.txt)
[serverlog.txt](https://github.com/debauchee/barrier/files/2188117/serverlog.txt)

---

> Commented on: 2018-09-08 by @walker0643

@mcx808 this should be fixed with 075d4f4758f86c49ef0b5257a2de076b1d7d4569. can you try it and report back? if you're not comfortable building barrier from source this fix will be included in the next release (2.2). thanks!

---

> Commented on: 2018-09-08 by @walker0643

Reopen if your issue isn't resolved after 2.2. Thank you :)

---

> Commented on: 2019-01-02 by @CaptainJawZ

Sorry for opening the issue again, but I am experimenting the same problem, with similar devices, here is my setup:
[Desktop] [Desktop}
[Surface pro]

When I use my desktop as a client it works seamlessly, just as I expect it.
However, when I try to run the surface as a client, the mouse just get's stuck on the bottom of the screen, top left corner.
I don't know if it's a DPI issue, or if it's a problem with multiple monitor setup, which I still havent figured out what's the right way to set, is it making two monitors with the same name? 

---

> Commented on: 2019-01-12 by @luckcolors

Hello.
I'm having the same issue. I'm willing to help you debug this, feel free to ask me things to test.

---

> Commented on: 2019-01-14 by @luckcolors

So, as a test i've built the latest version of the code from the repo and the fix you have committed https://github.com/debauchee/barrier/commit/075d4f4758f86c49ef0b5257a2de076b1d7d4569.
And it's working correctly so yeah this issue should be already fixed.
Please make an ufficial build whenever you get the chance. 😄 
Thanks for your help.

Also i've noticed that the setup build script points to a folder wich doesn't exist.

---

> Commented on: 2019-01-14 by @AdrianKoshka

> Also i've noticed that the setup build script points to a folder wich doesn't exist.

What do you mean? Could you get me log output from the build script?

---

> Commented on: 2019-01-14 by @luckcolors

The [line](https://github.com/debauchee/barrier/blob/4dedd88ab2fc900b44cea0003cddf065f01fa521/build_installer.bat#L9) in the build_installer.bat tries to CD into build/setup wich isn't created by the build script.
Instead two folders wich are named similarly are present: setup-wix, setup-inno.
I guess it's just a broken reference?

Log:
```
C:\barrier>build_installer.bat
Cannot find the specified file.
To build a 64-bit Windows installer:
 - set Q_BUILD_TYPE=Release in build_env.bat
 - also set other environmental overrides necessary for your build environment
 - run clean_build.bat to build Barrier and verify that it succeeds
 - re-run this script to create the installation package
```

---

> Commented on: 2019-01-14 by @luckcolors

Also i've noticed that the error mentions to change the variable Q_BUILD_TYPE=Release, but in the clean_build.cmd the variable is called B_BUILD_TYPE=Release.
Setting Q_BUILD_TYPE=Release does nothing.
(Also i did build successfully barrier by setting the proper variables before so no idea about this).

---

> Commented on: 2019-01-14 by @AdrianKoshka

did you read this? https://github.com/debauchee/barrier/wiki/Building-on-Windows Pretty sure you don't need to run `build_installer.bat`.

---

> Commented on: 2019-01-15 by @luckcolors

I didn't so that's probably why 😄 .

---

> Commented on: 2019-01-20 by @luckcolors

Turned on barrier again today, it broke again. Restarting barrier on both machines doesn't fix it.

---

> Commented on: 2019-01-20 by @AdrianKoshka

Oh, huh.

---

> Commented on: 2019-01-20 by @p12tic

In my experience sometimes barrier breaks the input system of either or both the server and the client. Restarting the machines themselves helps.

---

> Commented on: 2020-01-10 by @FredNass

Hello,

This happens on my side too with the following setup:

- Server : Linux Fedora 31 with barrier 2.3.2-snapshot-9080ce45 installed with Snap
- Client : macOS Catalina 10.15.2 with barrier 2.3.2 downloaded from Github

I installed 2.3.1 on a brand new Mac Mini running Mojave. I can swear it worked once (I could see and move the client's cursor by moving server's mouse) then the client's cursor disappeared forever. Actually, it got stucked in the bottom right corner of the client's screen.

Restarting Barrier on both server and client doesn't help.
Restarting both machines does not help either.
Uninstalling and reinstalling did not help
Upgrading macOS to Catalina did not help.

When both barriers and barrierc are started and connected, as soon as the server's cursor crosses the right screen edge (I use 2 screens on the server machine), it jumps to the client's bottom right of the screen and get stucked there.

The only way I can get my mouse back on the server is by stopping barrierc on the client.

Regards,
Frédéric.

---

> Commented on: 2020-04-23 by @emilpaw

Hello,
I have the same problem.
- Server : Windows 10 in a Dual Display Setup with barrier 2.3.2
- Client : Arch Linux with barrier 2.3.2

The problem only occurs if I have my second display connected. If I connect it, the cursor will change to the client and get stuck in the top right corner.
When I don't use my second display, everything runs fine.

My first display is an HiDPI one, so I doubt the HiDPI display itself is the root of the bug.

---

> Commented on: 2021-10-05 by @yume-chan

There was a fix for client mode in #306, not sure why it's not applied for server mode and why it was removed in aea488167af19845369e062bf9af4edfadc4fa71 as a "clean up".

Anyway, I found a workaround which you can apply now (I only have one monitor, I'm not sure whether it will work with multiple monitors):

1. Open Explorer and navigate to Barrier installation folder
2. Right-click on `barrierc.exe` or `barriers.exe` depends on your usage (client mode vs server mode), or both if you use both
3. Select "Properties"
4. Switch to "Compatibility" tab and click the "Change high DPI settings" button
    ![image](https://user-images.githubusercontent.com/1330321/136028803-b18b4320-509a-47f9-a6f0-6e926f1f5fb9.png)
5. In the new dialog, check "Override high DPI scaling behavior" and make sure "Application" is selected in the "Scaling performed by" dropdown menu
    ![image](https://user-images.githubusercontent.com/1330321/136029264-8ca43f84-17ab-405f-8071-41dd7efc063d.png)
6. Click "OK" twice to close both dialogs

Now it should work.

---

> Commented on: 2021-11-02 by @jiboncom

Having this problem with a Surface Pro 5 as the server and Barrier 2.4

---

> Commented on: 2021-11-04 by @paulrrogers

Confirmed this happened on Windows 10 20H2 (server) and MacOS 10.14.6 (client). The above High DPI workaround didn't help in my case.

**Comment by MakeNModify** [comment]
Don't know if that helps but how it works for me:

Working Setup:
   Server: PC Win 10 HiDPI with 150% scaling running Barrier 2.3.3
   Client: Linux Mint 20.3 no scaling Barrier 2.4.0 (2.3.3 also worked)

So apparently scaling was not an issue in 2.3.3 so if the server runs on this version it works. If i run 2.4.0 as a server my cursor get stuck in the corner once I try to cross to the client.

**Comment by sithlord48** [comment]
should be fixed by #1968

**Comment by shymega** [comment]
Again, let's also add a message to singular commit to #1968 like this

> Fixes #94

Thanks.

**Comment by sithlord48** [comment]
> Again, let's also add a message to singular commit to #1968 like this
> 
> > Fixes #94
> 
> Thanks.

My bad i thought it had landed already for some reason

**Comment by shymega** [comment]
No worries at all.
